From bcab7614602c068de34941fea730e168928e4bf1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Mika=20Bostr=C3=B6m?= <bostik@odin.bostik.iki.fi>
Date: Mon, 12 Dec 2011 09:11:04 +0200
Subject: [PATCH 03/13] Add build fixes to QtQuick3D imports

Provide the threed module header paths for both imports at build time.
Use the new includes.pri file.

Kill off DESTDIR declarations. We can not build directly to system
paths.

Kill off copyqmlinfra. We can not copy files to system paths during
build either.

Everything needs to go through install step!
---
 src/imports/shapes/shapes.pro |   12 +++---------
 src/imports/threed/threed.pro |   12 +++---------
 2 files changed, 6 insertions(+), 18 deletions(-)

diff --git a/src/imports/shapes/shapes.pro b/src/imports/shapes/shapes.pro
index 26d8ca2..2977014 100644
--- a/src/imports/shapes/shapes.pro
+++ b/src/imports/shapes/shapes.pro
@@ -2,9 +2,10 @@ TEMPLATE = lib
 TARGET  = qshapesqmlplugin
 CONFIG += qt plugin
 
-QT += declarative qt3d quick3d
+QT += declarative quick qt3d quick3d
+
+include(../../threed/includes.pri)
 
-DESTDIR = $$QT.quick3d.imports/Qt3D/Shapes
 target.path = $$[QT_INSTALL_IMPORTS]/Qt3D/Shapes
 INSTALLS += target
 
@@ -34,13 +35,6 @@ QML_INFRA_FILES += \
     teapot.bez
 # see the file README.library_xml for more on library.xml
 
-copyqmlinfra.input = QML_INFRA_FILES
-copyqmlinfra.output = $$target.path/${QMAKE_FILE_IN_BASE}${QMAKE_FILE_EXT}
-copyqmlinfra.commands = $$QMAKE_COPY ${QMAKE_FILE_IN} ${QMAKE_FILE_OUT}
-copyqmlinfra.CONFIG += no_link_no_clean
-copyqmlinfra.variable_out = PRE_TARGETDEPS
-QMAKE_EXTRA_COMPILERS += copyqmlinfra
-
 OTHER_FILES += \
     README.plugins_types \
     README.library_xml
diff --git a/src/imports/threed/threed.pro b/src/imports/threed/threed.pro
index 9151b7a..d21bd13 100644
--- a/src/imports/threed/threed.pro
+++ b/src/imports/threed/threed.pro
@@ -2,9 +2,10 @@ TEMPLATE = lib
 TARGET  = qthreedqmlplugin
 CONFIG += qt plugin
 
-QT += widgets opengl declarative qt3d quick3d
+QT += widgets opengl declarative quick qt3d quick3d
+
+include(../../threed/includes.pri)
 
-DESTDIR = $$QT.quick3d.imports/Qt3D
 target.path = $$[QT_INSTALL_IMPORTS]/Qt3D
 INSTALLS += target
 
@@ -33,13 +34,6 @@ QML_INFRA_FILES = \
     library.xml \
     plugins.qmltypes
 
-copyqmlinfra.input = QML_INFRA_FILES
-copyqmlinfra.output = $$[QT_INSTALL_IMPORTS]/Qt3D/${QMAKE_FILE_IN_BASE}${QMAKE_FILE_EXT}
-copyqmlinfra.commands = $$QMAKE_COPY ${QMAKE_FILE_IN} ${QMAKE_FILE_OUT}
-copyqmlinfra.CONFIG += no_link_no_clean
-copyqmlinfra.variable_out = PRE_TARGETDEPS
-QMAKE_EXTRA_COMPILERS += copyqmlinfra
-
 OTHER_FILES += \
     README.plugins_types \
     README.library_xml
-- 
1.7.7.3

